tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -487,6 +488,112 @@ def test_scontrol_reboot(
+    for key, val in clustermgtd_config.items():
+        remote_command_executor.run_remote_command(
+            f"sudo sed -i '/{key}/d' /etc/parallelcluster/slurm_plugin/parallelcluster_clustermgtd.conf"
+        )
+        remote_command_executor.run_remote_command(
+            f'echo "{key} = {val}" | sudo tee -a ' "/etc/parallelcluster/slurm_plugin/parallelcluster_clustermgtd.conf"
+        )
+
+    # Get compute nodes in the cluster
+    compute_nodes = slurm_commands.get_compute_nodes()
+
+    # Wait for compute nodes to be fully up and running
+    wait_for_compute_nodes_states(
+        slurm_commands,
+        compute_nodes,
+        "idle",
+        stop_max_delay_secs=300,
+    )
+
+    # Add drop-in file to delay startup of slurmd on the compute nodes. This will cause the compute node to
+    # stay longer in REBOOT_ISSUED state.
+    # TODO: generalize it for multiple nodes (it requires a parallel ssh utility such as clush).
+    # Currently the test uses only one compute node, so this is enough.
+    script = "add_sleep_to_slurmd_service_compute.sh"
+    remote_command_executor.run_remote_command(
+        f"rsync -a {script} {str(compute_nodes[0])}:~/", additional_files=[str(test_datadir / script)]
+    )
+    remote_command_executor.run_remote_command(f'ssh {str(compute_nodes[0])} "chmod +x {script}"')
+    remote_command_executor.run_remote_command(f'ssh {str(compute_nodes[0])} "./{script}"')
+
+    # Clear clustermgtd and slurmctld logs before starting the tests
+    remote_command_executor.run_remote_command("sudo truncate -s 0 /var/log/slurmctld.log")
+    remote_command_executor.run_remote_command("sudo truncate -s 0 /var/log/parallelcluster/clustermgtd")
+
+    # 2 iterations to cover the two scenarios described in the test description
+    for scenario in range(1, 3):
+
+        if scenario == 2:
+            # Remove delay in startup of slurmd on the compute nodes.
+            # TODO: generalize it for multiple nodes (it requires a parallel ssh utility such as clush)
+            # Currently the test uses only one compute node, so it's not necessary.
+            script = "reset_sleep_slurmd_service_compute.sh"
+            remote_command_executor.run_remote_command(
+                f"rsync -a {script} {str(compute_nodes[0])}:~/", additional_files=[str(test_datadir / script)]
+            )
+            remote_command_executor.run_remote_command(f'ssh {str(compute_nodes[0])} "chmod +x {script}"')
+            remote_command_executor.run_remote_command(f'ssh {str(compute_nodes[0])} "./{script}"')
+
+        # Reboot static compute nodes
+        for node in compute_nodes:
+            slurm_commands.reboot_compute_node(node, asap=True)
+
+        # Wait for compute nodes to reply to slurmctld after rebooting
+        wait_for_slurm_rebooted_nodes(compute_nodes, remote_command_executor, stop_max_delay_secs=600)
+
+        # Check that the nodes are not marked as unhealthy in the following minutes
```

Comment:
Can we add a check to verify the node fail health check?

Reply:
I agree with you, but here there's an annoying issue: it's not guaranteed that the EC2 checks will fail, because that depends on the time it takes for the reboot to occur. Sometimes a reboot is very quick and the EC2 status checks won't fail, and some other times the reboot might take longer, which causes the EC2 checks to fail.

Adding an assertion here may cause the test to fail.

In the end I decided to just verify at the end of the iteration that the node had not been drained during the reboot process.